### PR TITLE
logger.error added

### PR DESCRIPTION
**Pull request by olhabn**

### CODE REVIEW

File: src/foam/nanos/jetty/HttpServer.js [hunk 1]
```diff
@@ -209,7 +210,10 @@ foam.CLASS({
         this.configHttps(server);
                 
         server.start();
-      } catch(Exception e) {}
+      } catch(Exception e) {
+        Logger logger = (Logger) getX().get("logger");
+        logger.error(e);
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]: http service starts really early, so test for null logger here and the one below please. thanks.

File: src/foam/nanos/logger/FileLogger.js [hunk 2]
```diff
@@ -53,7 +53,10 @@ try {
 
   handler.setFormatter(new CustomFormatter());
   getLogger().addHandler(handler);
-} catch (IOException e) {}`
+} catch (IOException e) {
+  foam.nanos.logger.Logger logger = (foam.nanos.logger.Logger) getX().get("logger");
```
Comment (jlhughes) [inline src/foam/nanos/logger/FileLogger.js]: This has to be a system.err else we'll have a loop.  Can't use logging from within logging.

File: src/foam/nanos/jetty/HttpServer.js (hunk 1 above)
Comment (olhabn) [reply src/foam/nanos/jetty/HttpServer.js]: fixed

Comment (olhabn) [reply src/foam/nanos/jetty/HttpServer.js]: fixed

File: src/foam/nanos/logger/FileLogger.js (hunk 2 above)
Comment (olhabn) [reply src/foam/nanos/logger/FileLogger.js]: fixed

File: src/foam/nanos/jetty/HttpServer.js [hunk 3]
```diff
@@ -231,6 +236,8 @@ foam.CLASS({
               server.stop();
             } catch (Exception e) {
               System.err.println("Exception during Jetty server stop in the shutdown hook");
+              Logger logger = (Logger) getX().get("logger");
+              logger.error(e);
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]:
```suggestion
              if ( logger != null ) {
                logger.error(e);
              }
```

File: src/foam/nanos/jetty/HttpServer.js [hunk 4]
```diff
@@ -209,7 +210,11 @@ foam.CLASS({
         this.configHttps(server);
                 
         server.start();
-      } catch(Exception e) {}
+      } catch(Exception e) {
+        Logger logger = (Logger) getX().get("logger");
+        if(logger != null)
```
Comment (jlhughes) [inline src/foam/nanos/jetty/HttpServer.js]:
```suggestion
        if ( logger != null )
```

**Pull request merged**